### DISCUSSION TITLE
Is some documentation available?

### DISCUSSION THREAD

**Discussion by davide-giacomini** [francof2a/fxpmath#82] [Q&A]
Hello,

This is a question mostly directed to the developer. I am very happy about this library and it is helping me A LOT for an FPGA project.

I would like to know if there is actual documentation for the numerous methods available for conversion and so on. I feel like I could better harness this library, but I'm limiting myself to trial and error starting from the examples of the README.

Thank you so much in advance,
Take care,
Davide.

**Comment by francof2a** [comment]
Hello @davide-giacomini,

Thanks for using `fxpmath` and give some feedback!

Unfortunately, the only documentation I have done is the README file and some examples on the notebooks.

Clearly, it is a point to improve, but I have to optimize the little time I have available for this project, so it is something that I still cannot seriously do.

It would be nice to know what kind of documentation or examples you consider most important.

**Reply by davide-giacomini** [reply]
Hello  @francof2a,

Thank you for the answer.

I could give you an example: it took me a while to understand that I had the possibility to use `set_val(value, raw=True)` to use raw bits as input, and it took me even longer to understand that the value must be anyway cast to signed integer and it doesn't work if it is an unsigned integer (I mean, it works only for unsigned values). If there were a detailed documentation of the method `set_val` with all its possible parameters, that would be very useful.

I saw that you have your methods commented in the code. It would be enough to export those comments in a documentation that could be made in `readthedocs`, `GitHub`, or even a standard old documentation like `Java`'s.

Thank you again, hope this comment could be of help.
